src/_repobee/fileutil.py

```diff
@@ -0,0 +1,40 @@
+"""Utility functions for dealing with files and directories.
+
+.. module:: fileutil
+    :synopsis: Utility functions for dealing with files and directories.
+"""
+import enum
+import pathlib
+
+import repobee_plug as plug
+
+__all__ = ["DirectoryLayout"]
+
+
+def _flat_repo_path(
+    base: pathlib.Path, repo: plug.StudentRepo
+) -> pathlib.Path:
+    return base / repo.name
+
+
+def _by_team_repo_path(
+    base: pathlib.Path, repo: plug.StudentRepo
+) -> pathlib.Path:
+    return base / repo.team.name / repo.name
+
+
+class DirectoryLayout(enum.Enum):
+    """Layouts for arranging repositories on disk."""
+
+    FLAT = "flat"
+    BY_TEAM = "by-team"
+
+    def __init__(self, label: str):
+        self.label = label
+        self.get_repo_path = {
+            "flat": _flat_repo_path,
+            "by-team": _by_team_repo_path,
+        }[label]
```

Comment:
There are two options here: either attach the function to the enum itself, _or_ write a function like the one that previously existed and select a function based on the enum. The benefit to doing it _this_ way is that it's impossible to have an enum that does not have an associated function, because the `__init__` function is called when all of the enum instances are created, which happens when the class is loaded.

It's a little bit obtuse, but IMO a pretty neat solution to the problem of selecting functionality based on command line input.